lib/actions/setOrderNumber.js

```diff
@@ -0,0 +1,66 @@
+var sphere = require('../sphere.js');
+var messages = require('../messages.js');
+
+exports.process = action;
+
+/**
+ * Action for set order number
+ * @params {Object} msg
+ * @params {String} msg.orderId - order ID - required
+ * @params {Number} msg.orderNumber - order number to set - required
+ */
+function action(msg, cfg) {
+    var self = this;
+    var client = sphere.createServiceClient('orders', cfg);
+    var data = msg.body;
+
+    client
+        .byId(data.orderId)
+        .fetch()
+        .then(setOrderNumber)
+        .catch(handleError)
+        .finally(end)
+        .done();
+
+    function setOrderNumber(response) {
+        var order = response.body;
+        var orderNumber = data.orderNumber;
+        if (orderNumber === null || orderNumber === "") {
```

Comment:
What if orderNumber is undefined? I'd propose you to do just `if (!orderNumber)`. It will cover all these cases.